include/beman/any_view/detail/any_iterator.hpp

```diff
@@ -50,7 +68,13 @@ class any_iterator {
 
     constexpr any_iterator& operator=(any_iterator&&) noexcept = default;
 
-    [[nodiscard]] constexpr reference operator*() const { return **iterator_ptr; }
+    [[nodiscard]] constexpr reference operator*() const {
```

Comment:
If the `reference` is an actual reference and caching is enabled, the lifetime of the `*it` tied to `it` making it a stashing iterator. Stashing iterators have to be input iterators, so this can't work.

You can only use the optimization if the reference is not a reference type or if the view category is input.

Reply:
@foonathan If it's an actual reference, `it` does not own the reference, it simply stores the address. The lifetime of the reference isn't tied to the lifetime of the iterator. If it's a prvalue, `operator*` returns a prvalue as well (and the optimization is only enabled when that prvalue is trivially copyable).

I don't see how this is a stashing iterator, could you please clarify?

Reply:
@foonathan see the test I've added: https://github.com/bemanproject/any_view/pull/9/files#diff-830ac1a7c7e0b0b849871af7dcaf4360bdbcec98a97a81f03d81b2fc214b5d95